app/screens/post_options/post_options.js

```diff
@@ -63,9 +66,9 @@ export default class PostOptions extends PureComponent {
 
     getAddReactionOption = () => {
         const {formatMessage} = this.context.intl;
-        const {canAddReaction, channelIsReadOnly, isSystemMessage, showAddReaction} = this.props;
+        const {canAddReaction, showAddReaction} = this.props;
 
-        if (!isSystemMessage && showAddReaction && canAddReaction && !channelIsReadOnly) {
+        if (showAddReaction && canAddReaction) {
```

Comment:
can you please restore the previous condition? cause you cannot react to a system message. The `canReply` flag is set to false in the connector if the channel is read only but you forgot to check of system message. You can change it in the connector or here I'm 0/5 where but 5/5 on adding the condition.

```suggestion
        if (!isSystemMessage && showAddReaction && canAddReaction) {
```